### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by Goron1606** [SpiderLabs/ModSecurity-apache#46]
I'm using 
_ModSecurity v3.0.3 (Linux)_

I pulled latest commit for _modsecurity-apache_ just to test whether post data is accessible by _ARGS_POST_.

I'm getting the below error in apache error logs.

```
[Thu Nov 22 11:57:20.005976 2018] [core:notice] [pid 3768] AH00051: child pid 3823 exit signal Segmentation fault (11), possible coredump in /etc/apache2
[Thu Nov 22 11:57:20.015739 2018] [watchdog:debug] [pid 3832] mod_watchdog.c(565): AH02980: Watchdog: nothing configured?
[Thu Nov 22 11:57:20.016621 2018] [proxy:debug] [pid 3832] proxy_util.c(1785): AH00925: initializing worker proxy:reverse shared
[Thu Nov 22 11:57:20.016965 2018] [proxy:debug] [pid 3832] proxy_util.c(1827): AH00927: initializing worker proxy:reverse local
[Thu Nov 22 11:57:20.017503 2018] [proxy:debug] [pid 3832] proxy_util.c(1878): AH00931: initialized single connection worker in child 3832 for (*)
```

Please let me know what is going wrong.

_Originally posted by @Goron1606 in https://github.com/SpiderLabs/ModSecurity-apache/pull/22#issuecomment-440927889_

**Comment by c0deguard** [comment]
I'm also facing a similar issue.
I guess this because of

**Comment by c0deguard** [comment]
I'm also facing similar issue.
I guess this is because of some bug in "Proposed fix for msc_process_request_body call too early"
Here it shows building failed because of this [https://travis-ci.org/SpiderLabs/ModSecurity-apache](https://travis-ci.org/SpiderLabs/ModSecurity-apache)

**Comment by victorhora** [comment]
Hi all

I think there's an issue there indeed, but i'm not sure its related with the fact that the tests from Travis are failing. This is what Travis says about the build:

![image](https://user-images.githubusercontent.com/15767386/49181397-43b3da80-f325-11e8-90e1-a29e769bcdd4.png)

So Apache is failing to start (or the script is believes is the case) and unless the issue is happening right at the start of Apache/ModSecurity, it might be something wrong with the tests.

Could you please provide a full backtrace/stacktraces of this issue for further investigation?

Thanks

**Comment by Goron1606** [comment]
@zimmerle Can can please specify the logs which you require and their file paths?

**Comment by victorhora** [comment]
I would recommend recompiling everything (ModSecurity, Apache, Apache-connector) with debug symbols. This is usually accomplished by appending CFLAGS/CPPFLAGS like so:

```
$ export CFLAGS="-g -O0"
```

After that you can collect the stack traces by following these steps: https://httpd.apache.org/dev/debugging.html

You should also check the debugging section in the README of [libModSecurity ](https://github.com/SpiderLabs/ModSecurity/blob/v3/master/README.md)and the [Apache connector](https://github.com/SpiderLabs/ModSecurity-apache/blob/master/README.md)

Thanks for your support :)

**Comment by zimmerle** [comment]
Hi @Goron1606,

Do you happens to have **mod_unique_id** loaded?

- https://httpd.apache.org/docs/2.4/mod/mod_unique_id.html

**Comment by Goron1606** [comment]
Hi @zimmerle ,

I confirm the module is enabled.

**Comment by zimmerle** [comment]
Fixed at: 61f2ff14f690da6c05f0e7053bb41b430ae8b732

Thanks @Goron1606

**Comment by Goron** [comment]
@zimmerle ,

Thanks for the quick fix.
